mygraph.py

```diff
@@ -208,34 +226,48 @@ def __init__(self, directed: bool, n: int=0, simple: bool=False):
         self._e = list()
         self._simple = simple
         self._directed = directed
-        self._next_label_value = 0
+        self._tag = None
 
         for i in range(n):
-            self.add_vertex(Vertex(self))
+            self.add_vertex(Vertex())
 
     def __repr__(self):
         """
         A programmer-friendly representation of the Graph.
         :return: The string to approximate the constructor arguments of the `Graph'
         """
-        return 'Graph(directed={}, simple={}, #edges={n_edges}, #vertices={n_vertices})'.format(
-            self._directed, self._simple, n_edges=len(self._e), n_vertices=len(self._v))
+
+        tag_string = ''
+        if self._tag is not None:
+            tag_string = f'tag="{self._tag}", '
+
+        return \
+            'Graph(' \
+            + tag_string + \
+            'directed={}, ' \
+            'simple={}, ' \
+            '#edges={}, ' \
+            '#vertices={})'.format(self._directed, self._simple, len(self._e), len(self._v))
 
     def __str__(self) -> str:
         """
         A user-friendly representation of this graph
         :return: A textual representation of the vertices and edges of this graph
         """
-        return 'V=[' + ", ".join(map(str, self._v)) + ']\nE=[' + ", ".join(map(str, self._e)) + ']'
 
-    def _next_label(self) -> int:
-        """
-        Generates unique labels for vertices within the graph
-        :return: A unique label
-        """
-        result = self._next_label_value
-        self._next_label_value += 1
-        return result
+        tag_string = ''
+        if self._tag is not None:
+            tag_string = f'"{self._tag}"\n'
+
+        return \
+            tag_string + \
+            'V=[{0}]\n' \
+            'E=[{1}]'.format(", ".join(map(str, self._v)), ", ".join(map(str, self._e)))
+
+    def __eq__(self, other):
+        if isinstance(self, other.__class__):
```

Comment:
goed, had m zelf ook gemaakt alleen dan door de graphs per vertex te vergelijken

Reply:
Dit vond ik op StackOverflow ergens. Vond het wel een handige shortcut!

Maar we kunnen misschien beter overwegen zelf te bedenken wat de definitie is van graph object equivalence. Misschien is het vergelijken van vertices en edges al voldoende? Mijn methode vergelijkt de meeste attributes van een graafobject.

Reply:
Ja goed idee, maar ik denk dat dit in eerste instantie wel oke is. Misschien als we later aan het speedtunen zijn dat we er nog ff naar kunnen kijken.